### PR TITLE
Annotation.contains shouldn't consider the text+arrow's joint bbox.

### LINKED ISSUE #10875: Annotation.contains and FancyArrow.contains return incorrect values
### Bug report

Consider the following example:
```
from pylab import *
from matplotlib.backend_bases import MouseEvent
rcdefaults()

fig, axs = plt.subplots(2)
for i, ax in enumerate(axs):
    ax.set(xlim=(0, 1), ylim=(0, 1))
    ann = ax.annotate(
        "hello", xy=(.4, .4), xytext=(.6, .6), arrowprops={"arrowstyle": "->"})
    ax.figure.canvas.draw()
    xs = np.linspace(*ax.get_xlim(), 30)
    ys = np.linspace(*ax.get_ylim(), 30)
    for x in xs:
        for y in ys:
            event = MouseEvent(
                "button_press_event",
                ax.figure.canvas,
                *ax.transData.transform_point((x, y)))
            art = [ann, ann.arrow_patch][i]
            ax.plot(x, y, ".", c="g" if art.contains(event)[0] else "r")
plt.show()
```

figure_1

Points in green mark (a) in the top plot, points that are "contained" (per artist.contains) in the annotation object; (b) in the bottom plot, points that are "contained" in the arrow patch of the annotation object.

Obviously neither set of contained points is correct.  This affects, for example, the points that trigger interactive dragging of the annotation.

**Matplotlib version**

  * Operating system:
  * Matplotlib version: 2.2.1, but also present in 1.5
  * Matplotlib backend (`print(matplotlib.get_backend())`): qt5
  * Python version: 3.6
  * Jupyter version (if applicable):
  * Other libraries:

### PR DESCRIPTION
**Pull request by anntzer**
Previously, when deferring to Text.contains, self.get_window_extent
would incorrectly use Annotation.get_window_extent.  Just force it to
use Text.get_window_extent instead.

Also make Annotation obey the optional `._contains` attribute.

Also renamed the parameter to Annotation.contains to mouseevent, which is consistent with Text.contains and most other contains() methods.  I decided to skip the deprecation on that...

Closes #10875.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/text.py [hunk 1]
```diff
@@ -2181,12 +2184,13 @@ def transform(renderer) -> Transform
         else:
             self.arrow_patch = None
 
-    def contains(self, event):
-        contains, tinfo = Text.contains(self, event)
+    def contains(self, mouseevent):
```
Comment (timhoffm) [inline lib/matplotlib/text.py]: The change of the parameter name is an API change. While, naming of this parameter is not consistent throughout the library, I prefer not to sneek this in with the PR.

That should be handled in a separate PR and unify all `contains` signatures. It needs at least an API-change note (if  not a `@cbook._rename_parameter` deprecation).

Comment (anntzer) [reply lib/matplotlib/text.py]: reverted that change
